### PR TITLE
feat: Hack #10 - Use Webflow Interactions and cookies to play your lo…

### PR DESCRIPTION
**Pull request by aexshafii**
…ading animation only once per day

### CODE REVIEW

File: webflow-hacks/interactions-hacks/10.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// not needed if using cdn
+// import * as Cookies from 'js-cookie';
```
Comment (alexiglesias93) [inline webflow-hacks/interactions-hacks/10.ts]: @aexshafii You just have to add `"types": ["js-cookie"]` in the `tsconfig.json` file.
It'll automatically detect the types.

File: package.json [hunk 2]
```diff
@@ -25,12 +25,14 @@
...
+    "@types/js-cookie": "^3.0.1",
     "esbuild": "^0.14.0",
     "prettier": "^2.2.1",
     "typescript": "^4.2.3"
   },
   "dependencies": {
     "@finsweet/webflow-addons": "^1.0.8",
-    "highlight.js": "^10.7.1"
+    "highlight.js": "^10.7.1",
+    "js-cookie": "^3.0.1"
```
Comment (alexiglesias93) [inline package.json]: If the script is imported via CDN, you only need to install the @types, no need to install the package itself.

Comment (aexshafii) [reply package.json]: got it 👌

File: webflow-hacks/interactions-hacks/10.ts (hunk 1 above)
Comment (aexshafii) [reply webflow-hacks/interactions-hacks/10.ts]: 👌

**Comment by aexshafii** [comment]
Update: added types and removed js-cookie package in last commit

**Review by alexiglesias93 — APPROVED** [review]

**Comment by erikmejias** [comment]
I see changes to these files:
```
_code-blocks/helpers.js
_code-blocks/hljs.js
_code-blocks/webflow/embed-code.js
_code-blocks/zendesk/embed-code.js
```

Why is that? Could it be that you ran `npm run build`? 
To keep things separate, please remove these changes before merging.

**Comment by alexiglesias93** [comment]
> I see changes to these files:
> ```
> _code-blocks/helpers.js
> _code-blocks/hljs.js
> _code-blocks/webflow/embed-code.js
> _code-blocks/zendesk/embed-code.js
> ```
> 
> Why is that? Could it be that you ran `npm run build`? 
> To keep things separate, please remove these changes before merging.

It's probably due to the `tsc` command. We should update the `tsconfig.json` to only compile specific folders.

**Comment by aexshafii** [comment]
yes this happens when I run `tsc`

**Comment by erikmejias** [comment]
Hmm, are you using `tsc` on the root folder against the entire project? If that's the case, consider using it against the specific `.ts` file.

**Comment by aexshafii** [comment]
I always do it in the folder where the ts file is. Is that not how it should be done?

**Comment by erikmejias** [comment]
There's not a bad/good take, it depends on the use-case.
In this case, running `tsc` specifying the files would be ideal to prevent the TypeScript compiler from traversing all the files.
Doing so will prevent us having to write specific rules for custom folders and such.

Check the [documentation](https://www.typescriptlang.org/docs/handbook/compiler-options.html) for more on this.

**Comment by aexshafii** [comment]
okay will make sure to run it that way from now on

**Comment by erikmejias** [comment]
Please remember to remove the affected non-related files from these commits. 
It also happens on these PRs:
- [Hack 52](https://github.com/finsweet/fs-snippets/pull/6)
- [Hack 3](https://github.com/finsweet/fs-snippets/pull/8)
- [Hack 38](https://github.com/finsweet/fs-snippets/pull/9)

**Comment by aexshafii** [comment]
will be done by tonight

File: _code-blocks/webflow/embed-code.js [hunk 3]
```diff
@@ -1,47 +1,10 @@
...
...tart"?i.push(l[0].node):i.pop(),u(l.splice(0,1)[0])}return c+ae(n.substr(a))}var De=e=>{console.error(e)},rt=(e,...t)=>{console.log(`WARN: ${e}`,...t)},G=(e,t)=>{console.log(`Deprecated as of ${e}. ${t}`)},Be=ae,at=J,ot=Symbol("nomatch"),Un=function(e){let t=Object.create(null),n=Object.create(null),a=[],c=!0,i=/(^(<[^>]+>|\t|)+|\n)/gm,d="Could not find the language '{}', did you forget to load/include a language module?",r={disableAutodetect:!0,name:"Plain text",contains:[]},o={noHighlightRe:/^(no-?highlight)$/i,languageDetectRe:/\blang(?:uage)?-([\w-]+)\b/i,classPrefix:"hljs-",tabReplace:null,useBR:!1,languages:null,__emitter:Ze};function u(s){return o.noHighlightRe.test(s)}function l(s){let h=s.className+" ";h+=s.parentNode?s.parentNode.className:"";let R=o.languageDetectRe.exec(h);if(R){let A=F(R[1]);return A||(rt(d.replace("{}",R[1])),rt("Falling back to no-highlight mode for this block.",s)),A?R[1]:"no-highlight"}return h.split(/\s+/).find(A=>u(A)||F(A))}function y(s,h,R,A){let I="",ee="";typeof h=="object"?(I=s,R=h.ignoreIllegals,ee=h.language,A=void 0):(G("10.7.0","highlight(lang, code, ...args) has been deprecated."),G("10.7.0",`Please use highlight(code, options) instead.
```
Comment (alexiglesias93) [inline _code-blocks/webflow/embed-code.js]: This file shouldn't be generated if you just compile the hack file.

File: _code-blocks/zendesk/embed-code.js [hunk 4]
```diff
@@ -1,45 +1,10 @@
...
...tart"?i.push(l[0].node):i.pop(),u(l.splice(0,1)[0])}return c+ae(n.substr(a))}var De=e=>{console.error(e)},rt=(e,...t)=>{console.log(`WARN: ${e}`,...t)},G=(e,t)=>{console.log(`Deprecated as of ${e}. ${t}`)},Be=ae,at=J,ot=Symbol("nomatch"),Un=function(e){let t=Object.create(null),n=Object.create(null),a=[],c=!0,i=/(^(<[^>]+>|\t|)+|\n)/gm,d="Could not find the language '{}', did you forget to load/include a language module?",r={disableAutodetect:!0,name:"Plain text",contains:[]},o={noHighlightRe:/^(no-?highlight)$/i,languageDetectRe:/\blang(?:uage)?-([\w-]+)\b/i,classPrefix:"hljs-",tabReplace:null,useBR:!1,languages:null,__emitter:Ze};function u(s){return o.noHighlightRe.test(s)}function l(s){let h=s.className+" ";h+=s.parentNode?s.parentNode.className:"";let S=o.languageDetectRe.exec(h);if(S){let w=F(S[1]);return w||(rt(d.replace("{}",S[1])),rt("Falling back to no-highlight mode for this block.",s)),w?S[1]:"no-highlight"}return h.split(/\s+/).find(w=>u(w)||F(w))}function y(s,h,S,w){let I="",ee="";typeof h=="object"?(I=s,S=h.ignoreIllegals,ee=h.language,w=void 0):(G("10.7.0","highlight(lang, code, ...args) has been deprecated."),G("10.7.0",`Please use highlight(code, options) instead.
```
Comment (alexiglesias93) [inline _code-blocks/zendesk/embed-code.js]: Same for this one.

Comment (aexshafii) [reply _code-blocks/zendesk/embed-code.js]: I just reverted the files to how they were on the main branch

Comment (alexiglesias93) [reply _code-blocks/zendesk/embed-code.js]: Got it, thanks

**Comment by aexshafii** [comment]
ready for review

File: webflow-hacks/interactions-hacks/10.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  const clearCookiesButton = document.querySelector<HTMLElement>(CLEAR_COOKIES_BUTTON_SELECTOR);
+
+  if (!clearCookiesButton) return;
+  // when  clearCookiesButton is clicked
+  clearCookiesButton.onclick = () => {
+    // remove the 'seenGif' cookie
+    // the animation can now play again since
+    //'seenAnimation' becomes undefined
+    Cookies.remove('seenAnimation');
+  };
```
Comment (alexiglesias93) [inline webflow-hacks/interactions-hacks/10.ts]: If this should not be added by the user, I would remove it.
I would put this code inside a separate file, something like `10-internal.ts`

**Review by alexiglesias93 — APPROVED** [review]

**Pull request merged**